docs/content/testing/writing-a-new-test.md

```diff
@@ -0,0 +1,344 @@
+ If it does not exist, please refer to the
+ [Creating a New Test Folder and Assembly](#creating-a-new-test-folder-and-assembly)
+ heading.
+
+### Test fixtures
+
+Test Fixtures allow test methods within them to share similar setups.
+
+Since this class has both static and non-static methods, they can be tested
+ within two separate fixtures.
+
+The setup/teardown logic are done through these annotations:
+
+- `[SetUp]` and `[TearDown]` methods are performed before and after each test.
+- `[OneTimeSetUp]` and `[OneTimeTearDown]` methods are peformed before and
+ after all tests.
+
+For each fixture, the execution happens in this order:
+
+Before any test has run:
+- `[OneTimeSetUp]`
+
+Then, e.g. if you have two tests:
+- `[SetUp]`
+- Test 1
+- `[TearDown]`
+- `[SetUp]`
+- Test 2
+- `[TearDown]`
+
+And after all tests have been run:
+- `[OneTimeTearDown]`
+
+The namespace for the fixture should be
+ `Improbable.Gdk.Core.EditmodeTests.Utility`, because the class you are testing
+ is in the `Improbable.Gdk.Core.Utility` namespace.
+
+```cs
+using Improbable.Gdk.Core.Utility;
+using NUnit.Framework;
+
+namespace Improbable.Gdk.Core.EditmodeTests.Utility
+{
+    [TestFixture]
+    public class MyClassStaticTests
+    {
+    }
+
+    [TestFixture]
+    public class MyClassInstanceTests
+    {
+        private MyClass myClassInstance;
+
+        [SetUp]
+        public void SetUp()
+        {
+            myClassInstance = new MyClass();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            myClassInstance.Dispose();
+        }
+    }
+}
+```
+
+### Test methods
+
+Let's write a few test methods.
+
+The test should ensure that the `AddTwoNumbers` static method returns the sum of the
+ input numbers.
+
+The test name should match this expectation and contain relevant assertions.
+
+```cs
+    [TestFixture]
+    public class MyClassStaticTests
+    {
+        [Test]
+        public void AddTwoNumbers_should_return_the_sum_of_the_input_numbers()
+        {
+            Assert.AreEqual(4, MyClass.AddTwoNumbers(1, 3));
+            Assert.AreEqual(4, MyClass.AddTwoNumbers(3, 1));
+            Assert.AreEqual(7, MyClass.AddTwoNumbers(2, 5));
+        }
+    }
+```
+
+For the instance functionality, you can add tests similarly:
+
+```cs
+    [TestFixture]
+    public class MyClassInstanceTests
+    {
+        private MyClass myClassInstance;
+
+        [SetUp]
+        public void SetUp()
+        {
+            myClassInstance = new MyClass();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            myClassInstance.Dispose();
+        }
+
+        [Test]
+        public void InstanceValue_is_0_by_default()
+        {
+            Assert.AreEqual(0, myClassInstance.InstanceValue);
+        }
+
+        [Test]
+        public void DoInstanceOperation_increases_InstanceValue_by_1()
+        {
+            myClassInstance.DoInstanceOperation();
+
```

Comment:
would remove the new lines between function calls and asserts to group them together